test/custom_operator/test_custom_ops_annotation.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+import unittest
+
+import torch
+from torch.testing._internal.common_utils import TestCase
+
+class TestCustomOperatorsWithAnnotation(TestCase):
+    def test_tensor(self):
+        @torch.library.custom_op("mylibrary::foo_op", mutates_args={})
+        def foo_op(x: torch.Tensor) -> torch.Tensor:
+            return x.clone()
+
```

Comment:
Could we test some other cases?
- we also accept int/List/Optional/float/torch.dtype/torch.device/torch.types.Number in schema
- what happens if we do `from torch import dtype` and use `dtype` instead of torch.dtype?
- we should test a failure case: use self.assertRaisesRegex

Reply:
Working on adding more test cases. 

For 2, it will still work, because we have `from .. import device, dtype, Tensor, types`, so `dtype` can be evaluated. However, if someone decided to do something like `from some_other_library import dtype`, then the current code will incorrectly evaluate that to torch.dtype instead of `some_other_library.dtype`. 

Unfortunately I don't see how we can prevent this problem after thinking about it for a while. Maybe we can alert the users about this.

Reply:
I'm not sure if there's a better way to do this. The best-effort approach in this PR sounds fine to me.